File: ruby/bob/bob.rb
module BookKeeping
  VERSION = 1
end

class Bob
  def self.hey(str)
    str.gsub!(/\s/, '')
    return 'Fine. Be that way!' if str == ''
    return 'Whoa, chill out!' if str == str.upcase && !str.match(/^[^a-zA-Z]+$/)
    return 'Sure.' if str[-1] == '?'
    'Whatever.'
  end
end

File: javascript/bob/bob.js
function Bob() {}

Bob.prototype.hey = function(string){
  var prompt = string.trim();
  
  if (prompt === "") return "Fine. Be that way!";
  if (prompt === prompt.toUpperCase() && !prompt.match(/^[^a-zA-Z]+$/)) return "Whoa, chill out!";
  if (prompt.slice(-1) === '?') return "Sure.";
  return "Whatever.";
  
};

module.exports = Bob;